### ISSUE TITLE
help with configuration authorization of buildbot

### ISSUE THREAD

**Issue by EndrII** [buildbot/buildbot#5146]
hi 
I need to configure github authorization. so that all team members
quasarappcore could use only 2 of 5 builders (github-worker and github-tester) and could only put on rebuild and build. If the user put the project on the assembly, then he, as the owner, could stop it.

my current configuration :

``` python
        self.masterConf['www']['authz'] = util.Authz(
                allowRules=[
                    util.AnyEndpointMatcher(role="admins"),
                    util.StopBuildEndpointMatcher(builder="*", role="owner"),
                    util.ForceBuildEndpointMatcher(builder="github*",
                                                   role="quasarcore"),
                    util.RebuildBuildEndpointMatcher(builder="github*",
                                                     role="quasarcore")
                ],
                roleMatchers=[
                  util.RolesFromGroups(groupPrefix='QuasarApp/'),
                  util.RolesFromEmails(admins=["EndrIIMail@gmail.com"]),
                  util.RolesFromOwner(role="owner")

                ]
        )

        secret = SecretManager("/home/andrei/buildBotSecret/secret.json")

        self.masterConf['www']['auth'] = util.GitHubAuth(
            secret.getValue("QuasarAppCIID"),
            secret.getValue("QuasarAppCIToken"),
            apiVersion=4, getTeamsMembership=True) 
```

my current builders:

``` python
        self.masterConf['workers'] = [
            worker.Worker("github-worker", "pass"),
            worker.Worker("github-tester", "pass"),
            worker.Worker("qtBuilder", "pass"),
            worker.Worker("NPM", "pass"),
            worker.Worker("assets-builder", "pass")

            ]
```

**Comment by EndrII** [comment]
Does anyone know what could be the problem?

**Comment by rvanlaar** [comment]
@EndrII What are you using for the JSON secrets manager?

**Comment by EndrII** [comment]
@rvanlaar you can see [here]
(https://github.com/QuasarApp/QuasarAppCI/blob/master/BuildBotLib/secretManager.py) 
but the problem is definitely not in it, I have been using it for a long time and all the data is returned correctly.

**Comment by rvanlaar** [comment]
@EndrII Yeah, sorry. no I can't help you with your problem. I recently rolled my own custom secret manager thing. Yours is way cooler.

**Comment by EndrII** [comment]
@rvanlaar thanks )
I will hope that someone will still help to figure this out.

**Comment by tardyp** [comment]
> my current builders:

Those are not builders, those are workers. Do you have 1:1 mapping between worker and builder?

Actually I read your report twice, I can't figure out exactly the question.

**Comment by EndrII** [comment]
@tardyp yes. I read my code now and found some errors, in particular builders and workers are mixed up. Now I do not have time to do this, but later I will do it and if there are difficulties or the result, I will unsubscribe here. In case of difficulties, I hope for your help.

**Comment by EndrII** [comment]
I checked many options and came to the conclusion that AnyControlEndpointMatcher ForceBuildEndpointMatcher StopBuildEndpointMatcher RebuildBuildEndpointMatcher is not working.
working only AnyEndpointMatcher.

now I will post the authorization files from different commits and sign which option worked and which one did not. 

this is [works](https://github.com/QuasarApp/QuasarAppCI/blob/master/BuildBotLib/buildBotIdentity.py) case - all users with the QuasarApp group have access to all builders.

but in next cases all users do not have any permisions.
* https://github.com/QuasarApp/QuasarAppCI/blob/a238c453c66e1c695d4b7f430096609727d485c7/BuildBotLib/buildBotIdentity.py
* https://github.com/QuasarApp/QuasarAppCI/blob/954befe47c20ca14d3a978975d27c3a904e8e3ac/BuildBotLib/buildBotIdentity.py
* https://github.com/QuasarApp/QuasarAppCI/blob/07d97fe544a30233c6ee4ce07ebaf96685098dd7/BuildBotLib/buildBotIdentity.py

It looks like a bug ...
Because of this behavior, I had to give all users of the group a administrator rights, which is not good.

**Comment by EndrII** [comment]
@tardyp @rvanlaar any ideas how to make all builds logs visible for all but not add access to control of builds.

**Comment by rvanlaar** [comment]
Our buildbot is open for the world, but only logged in users can kick off builds. 

https://github.com/scummvm/scummvm-sites/blob/director-buildbot/master.py#L191

```python
c["www"]["auth"] = util.GitHubAuth(
    env["GITHUB_CLIENT_ID"],
    env["GITHUB_CLIENT_SECRET"],
    apiVersion=4,
    getTeamsMembership=True,
)

c["www"]["authz"] = util.Authz(
    allowRules=[util.AnyControlEndpointMatcher(role="developers"),],
    roleMatchers=[util.RolesFromGroups(groupPrefix="scummvm/")],
)
```
